modules/DMZ/data_kit/data_filler.py

```diff
@@ -1,18 +1,27 @@
 import pandas as pd
 import numpy as np
+import dataset_insight
 
 
-# Drops rows which have fewer than threshold observations.
-# e.g. if threshold = 5, any rows with fewer than 5 observations will be removed from pandas_data.
-def drop_missing_data_rows(pandas_data, threshold):
-    pandas_data = pandas_data.dropna(thresh=threshold)
+# Drops rows which have a missing to existing component ratio greater than the desired_ratio.
+# e.g. if desired_Ratio is .5, and a row is missing 60% of its components, the row will be dropped.
+def drop_missing_data_rows(pandas_data, desired_ratio):
```

Comment:
@ASAAR I tried to merge the two together, but didn't have much success because of the way that pandas handles rows and columns differently. I feel like there's a way to do it, but I spent about two hours trying different things and had no luck.